server/server.go

```diff
@@ -42,6 +42,11 @@ type StartSettings struct {
 
 	// Server's TLS configuration.
 	TLSConfig *tls.Config
+
+	// HTTPMiddleware specifies middleware for HTTP messages received by the server.
+	// Note that the function will be called once for websockets upon connecting and will
+	// be called for every HTTP request. This function is optional to set.
+	HTTPMiddleware func(handler http.Handler) http.Handler
```

Comment:
I changed the middleware setting to be `func(handler http.Handler) http.Handler` from `func(handler http.HandlerFunc) http.HandlerFunc` because I'm finding that lots of middleware examples – including the middleware created by `otelhttp.NewMiddleware` – follow the `Handler -> Handler` pattern.

To make this change work, I ended up renaming the `server.httpHandler` function in _serverimpl` to `server.ServeHTTP` so that it conforms to the `http.Handler` interface.